src/pytorch_lightning/core/saving.py

```diff
@@ -171,13 +173,15 @@ def on_hpc_load(self, checkpoint: Dict[str, Any]) -> None:
 
 
 def _load_from_checkpoint(
-    cls: Union["pl.LightningModule", "pl.LightningDataModule"],
+    cls: Union[Type["ModelIO"], Type["pl.LightningModule"], Type["pl.LightningDataModule"]],
```

Comment:
these were annotated as types and not instances because 1) consistent annotations (not mixing Types and instances) across methods, and 2 mypy raises errors in situations such as 

```python
# ln 218
cls_spec = inspect.getfullargspec(cls.__init__)
```

because __init__ cannot be accessed (according to mypy) when cls is annotated as an instance, and not as a Type